### PR TITLE
nsd: 4.1.15 -> 4.1.16

### PR DESCRIPTION
**Pull request by leenaars**
###### Motivation for this change

Minor version update. Plus (in a second commit) a fix to nsd-control-setup. This convenience script  was dependent on openssl being already installed, now it is a propagatedBuildInput.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/servers/dns/nsd/default.nix [hunk 1]
```diff
@@ -13,14 +13,19 @@
...
+    sha256 = "1cmaddfjb7yr87gjd5yv4d0qng0j97sy5rw5m3zxsp6c4fnng0vz";
   };
 
-  buildInputs = [ libevent openssl ];
+  patchPhase = ''
+    substituteInPlace nsd-control-setup.sh.in --replace openssl ${openssl}/bin/openssl
+    '';
+
+  buildInputs = [ libevent ];
+  propagatedBuildInputs = [ openssl ];
```
Comment (fpletz) [inline pkgs/servers/dns/nsd/default.nix]: If you're already replacing the openssl path in the shell script calling it, putting it in `propagatedBuildInputs` is not necessary except there are more scripts/tools depending on openssl being present.

File: pkgs/servers/dns/nsd/default.nix [hunk 2]
```diff
@@ -13,14 +13,19 @@
...
+  name = "nsd-4.1.16";
 
   src = fetchurl {
     url = "http://www.nlnetlabs.nl/downloads/nsd/${name}.tar.gz";
-    sha256 = "494a862cfcd26a525a4bf06306eb7ab0387b34678ac6d37767507438e3a23a4b";
+    sha256 = "1cmaddfjb7yr87gjd5yv4d0qng0j97sy5rw5m3zxsp6c4fnng0vz";
   };
 
-  buildInputs = [ libevent openssl ];
+  patchPhase = ''
```
Comment (fpletz) [inline pkgs/servers/dns/nsd/default.nix]: You should use `prePatch` or `postPatch` here so you can still pass `patches` to the `mkDerivation` call which is done in the default `patchPhase` that you're replacing here.

File: pkgs/servers/dns/nsd/default.nix [hunk 3]
```diff
@@ -13,14 +13,19 @@
...
+    substituteInPlace nsd-control-setup.sh.in --replace openssl ${openssl}/bin/openssl
+    '';
```
Comment (fpletz) [inline pkgs/servers/dns/nsd/default.nix]: Minor indention nitpick. 2 spaces to the left please. :)

**Comment by leenaars** [comment]
Fixed comments by @fpletz

**Comment by hrdinka** [comment]
Built and tested locally. Thank you for your contribution!

**Pull request closed without merging**